core/src/main/java/fr/sncf/osrd/new_infra/implementation/tracks/undirected/UndirectedInfraBuilder.java

```diff
@@ -0,0 +1,173 @@
+        var edge = new InfraTrackSection(track.length, track.id);
+        builder.addEdge(begin, end, edge);
+        edge.getAttrs().putAttr(TRACK_OBJECTS, new ArrayList<>());
+        return edge;
+    }
+
+    /** Creates a node and registers it in the graph */
+    private void addNode(String trackId, EdgeEndpoint endpoint, TrackNode node) {
+        var map = beginEndpoints;
+        if (endpoint == EdgeEndpoint.END)
+            map = endEndpoints;
+        if (map.containsKey(trackId)) {
+            if (map.get(trackId) instanceof SwitchPort) // the link is already created in a switch, ignore
+                return;
+            throw new InvalidInfraException("Duplicated track link"); // TODO: add details
+        }
+        map.put(trackId, node);
+        builder.addNode(node);
+    }
+
+    /** Returns the node at the given end of the track */
+    private TrackNode getNode(String trackId, EdgeEndpoint endpoint) {
+        var map = beginEndpoints;
+        if (endpoint == EdgeEndpoint.END)
+            map = endEndpoints;
+        if (map.containsKey(trackId))
+            return map.get(trackId);
+        var res = new InfraTrackNode.End();
+        addNode(trackId, endpoint, res);
+        return res;
+    }
+
+    /** Parse a given RJS switch */
+    private Switch parseSwitch(
+            RJSSwitch rjsSwitch,
+            HashMap<String, RJSSwitchType> switchTypeMap
+    ) throws InvalidInfraException {
+        var networkBuilder = NetworkBuilder.directed()
+                .<SwitchPort, SwitchBranch>immutable();
+        var portMap = ImmutableMap.<String, SwitchPort>builder();
+        var allPorts = new HashSet<InfraSwitchPort>();
+        for (var entry : rjsSwitch.ports.entrySet()) {
+            var portName = entry.getKey();
+            var port = entry.getValue();
+            var newNode = new InfraSwitchPort(portName);
+            portMap.put(portName, newNode);
+            networkBuilder.addNode(newNode);
+            addNode(port.track.id.id, port.endpoint, newNode);
+            allPorts.add(newNode);
+        }
+        var finalPortMap = portMap.build();
+        var switchType = rjsSwitch.switchType.getSwitchType(switchTypeMap);
+        var groups = ImmutableMultimap.<String, SwitchBranch>builder();
+        var allBranches = new HashSet<InfraSwitchBranch>();
+        for (var entry : switchType.groups.entrySet()) {
+            for (var e : entry.getValue()) {
+                var src = finalPortMap.get(e.src);
+                var dst = finalPortMap.get(e.dst);
+                var branch = new InfraSwitchBranch(0);
+                groups.put(entry.getKey(), branch);
+                assert src != null;
+                assert dst != null;
+                networkBuilder.addEdge(src, dst, branch);
+                builder.addEdge(src, dst, branch);
+                allBranches.add(branch);
+                branch.getAttrs().putAttr(TRACK_OBJECTS, new ArrayList<>());
```

Comment:
It's more convenient if we can iterate over all the objects on the edge without having to check if the list exists.

But I can change it into a `Collections.emptyList()`, that way it will throw if we actually try to add objects on the branch.

Reply:
I agree this attribute should be removed, and that we will not use this "feature", but I don't see why we should forbid this case. if switch branches are edges, they must share some of its properties, such as the ability to have attributes, at least in theory.